components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1805,1229 +1180,654 @@ subroutine tphysac (ztodt,   cam_in,  &
+                              crm_ecpp_output%massflxbnd, crm_ecpp_output%rhcen,        &
+                              crm_ecpp_output%qcloudcen,  crm_ecpp_output%qlsinkcen,    &
+                              crm_ecpp_output%precrcen,   crm_ecpp_output%precsolidcen, &
+                              acldy_cen_tbeg)
+      call physics_update(state, ptend, dtstep_pp, tend)
+      call t_stopf ('ecpp')
+
+    end if ! nstep.ne.0 .and. mod(nstep, necpp).eq.0
+
+    call crm_ecpp_output%finalize()
+
+  end if ! use_ECPP
 #endif /* ECPP */
+  !-----------------------------------------------------------------------------
+  ! save old CRM cloud fraction - w/o CRM, this is done in cldwat2m.F90
+  !-----------------------------------------------------------------------------
+  call pbuf_get_field(pbuf,pbuf_get_index('CLDO'),cldo,start=(/1,1,itim_old/),kount=(/pcols,pver,1/))
+  call pbuf_get_field(pbuf,pbuf_get_index('CLD') ,cld ,start=(/1,1,itim_old/),kount=(/pcols,pver,1/))
 
-      !-------------------------------------------------------------------------
-      ! save old CRM cloud fraction - w/o CRM, this is done in cldwat2m.F90
-      !-------------------------------------------------------------------------
-
-      ifld = pbuf_get_index('CLDO')
-      call pbuf_get_field(pbuf, ifld, cldo, start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
-
-      ifld = pbuf_get_index('CLD')
-      call pbuf_get_field(pbuf, ifld, cld , start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
-
-      cldo(1:ncol,1:pver) = cld(1:ncol,1:pver)
-
-      !-------------------------------------------------------------------------
-      !-------------------------------------------------------------------------
-   end if ! use_MMF
-   !================================================================================================
-   !------------------------------------------------------------------------------------------------
-   !================================================================================================
-
-   if (l_tracer_aero) then
-      if (use_MMF.and.use_ECPP.and.MMF_microphysics_scheme.eq.'m2005') then
-         ! With 2-mom MMF+ ECPP we can skip the conventional aerosol routines.
-         ! With 1-mom MMF we use prescribed or bulk aerosols so the conventional
-         ! convective transport and wet savenging are still needed.
-      else if ( .not. deep_scheme_does_scav_trans() ) then
-
-         !======================================================================
-         ! Aerosol wet chemistry determines scavenging and transformations.
-         ! This is followed by convective transport of all trace species except
-         ! water vapor and condensate. Scavenging needs to occur prior to
-         ! transport in order to determine interstitial fraction.
-         !======================================================================
-
-         call t_startf('tphysbc_aerosols')
```

Comment:
I think it's worthwhile to keep the timers around this stuff so it's easier to reconcile the total cost of tphysbc with the individual calls.